db/log_reader.cc

```diff
@@ -313,11 +313,13 @@ bool Reader::ReadRecord(Slice* record, std::string* scratch,
         break;
 
       default: {
-        std::string reason =
-            "unknown record type " + std::to_string(record_type);
-        ReportCorruption(
-            (fragment.size() + (in_fragmented_record ? scratch->size() : 0)),
-            reason.c_str());
+        if (record_type <= kRecordTypeSafeIgnoreMask) {
```

Comment:
If it's a "mask", we should check it as a bit mask rather than a minimum value. Unless we have a sense of which is going to be most common, I suggest we split the remaining value space roughly in half for ignorable vs. rejected when unknown.

Reply:
Fixed